### PR TITLE
Fix the value of the rank if the interface is logical

### PR DESCRIPTION
**Pull request by CamilleEviden**
We did not setup the rank if the interface was logical

### CODE REVIEW

**Comment by traxys** [comment]
Please use commit title that describes the issue, 'Fix <the issue>'

**Comment by boyerq** [comment]
Please use a more explicit commit message, of the form `Fix <the issue>`

File: swptl/swptl.c [hunk 1]
```diff
@@ -3103,6 +3116,14 @@ int swptl_rcv_rstart(struct swptl_ni *ni, struct swptl_reply *reply, int nid, in
...
 		ptl_panic("%d: bad rank\n", reply->serial);
 
+	/* if (!SWPTL_ISPHYSICAL(ni->vc) && f->conn->rank == -1) {
+		f->conn->rank = swptl_ni_p2l(ni, nid, pid);
+		if (f->conn->rank == -1) {
+			LOGN(2, "%s: rank not set\n", __func__);
+			return 0;
+		}
+	} */
+
```
Comment (boyerq) [inline swptl/swptl.c:3126]: Why is this code commented?

File: swptl/swptl.c [hunk 2]
```diff
@@ -1274,6 +1274,14 @@ void swptl_trig(struct swptl_ni *ni)
...
 			sodata->conn = bximsg_getconn(ni->dev->iface, trig->u.tx.nid,
 						      trig->u.tx.pid, ni->vc);
+
+			if (!SWPTL_ISPHYSICAL(ni->vc)) {
+				sodata->conn->rank =
+					swptl_ni_p2l(ni, sodata->conn->nid, sodata->conn->pid);
+				if (sodata->conn->rank == -1) {
+					LOGN(2, "%s: rank not set\n", __func__);
+				}
+			}
```
Comment (boyerq) [inline swptl/swptl.c:1284]: Why in this branch, and not the other branches?

File: swptl/swptl.c [hunk 3]
```diff
@@ -3103,6 +3108,14 @@ int swptl_rcv_rstart(struct swptl_ni *ni, struct swptl_reply *reply, int nid, in
 	if (!SWPTL_ISPHYSICAL(ni->vc) && f->conn->rank == -1)
 		ptl_panic("%d: bad rank\n", reply->serial);
 
+	/* if (!SWPTL_ISPHYSICAL(ni->vc) && f->conn->rank == -1) {
+		f->conn->rank = swptl_ni_p2l(ni, nid, pid);
+		if (f->conn->rank == -1) {
+			LOGN(2, "%s: rank not set\n", __func__);
+			return 0;
+		}
+	} */
```
Comment (boyerq) [inline swptl/swptl.c:3117]: There is still commented code

**Comment by boyerq** [comment]
Could you change the commit title to the same name as the PR?

**Comment by boyerq** [comment]
I seen there are three calls to `bximsg_getconn`, one in swptl_cmd when `trig_ct != NULL`, one in `swptl_rcv_qstart` that already has a call to `swptl_ni_p2l`, and one in `swptl_trig`. Is there a reason we don't need a `swptl_ni_p2l` in each of those call sites?

**Comment by boyerq** [comment]
Could you rename the commit message to the PR title?

**Review by boyerq — APPROVED** [review]

**Pull request merged**